app/routes.php

```diff
@@ -282,6 +283,31 @@ function( Request $request ) use ( $app, $ffFactory ) {
 	}
 );
 
+$app->get(
+	'add-comment',
+	function( Request $request ) use ( $app, $ffFactory ) {
+		$template = $ffFactory->getLayoutTemplate(
+			'Donation_Comment.html.twig'
+		);
+
+		return new Response(
+			$template->render(
+				[
+					'donationId' => (int)$request->query->get( 'donationId', '' ),
+					'updateToken' => $request->query->get( 'updateToken', '' ),
+					'cancelUrl' => $app['url_generator']->generate(
+						'show-donation-confirmation',
+						[
+							'id' => (int)$request->query->get( 'donationId', '' ),
+							'accessToken' => $request->query->get( 'accessToken', '' )
+						]
+					)
+				]
+			)
+		);
+	}
+);
```

Comment:
To make the `UrlGenerator` work in `DonationConfirmationHtmlPresenter` you need to add 

    ->bind( 'add-comment' );

`UrlGenerator` expects a route name, not a route URL.

Reply:
Ahhh.... this caused quite some confoosion for me yesterday. Great you spotted this without me even asking why it's not working.